### PR TITLE
Fix huge gene lists for big cancer SVs in case report

### LINKED ISSUE #3913: HUGE gene lists shown on case the report
When a very big SV variant is involved. Solution:

- Show all genes if they are below a certain number (how many?)
- Filter these genes by gene panel (if available) if they are more than ?
- Don't show any gene if the variant contains A LOT (How many?) genes and no panel is available (but mention nr. of genes contained in the variant)

### PR DESCRIPTION
**Pull request by northwestwitch**
Tentative fix #3913 . test with [this case](https://scout-stage.scilifelab.se/cust000/TEST-SV-MAXSIZE)

<details>
<summary>Testing on cg-vm1 server (Clinical Genomics Stockholm)</summary>

**Prepare for testing**
1. Make sure the PR is pushed and available on [Docker Hub](https://hub.docker.com/repository/docker/clinicalgenomics/scout-server-stage)
1. Fist book your testing time using the Pax software available at [https://pax.scilifelab.se/](https://pax.scilifelab.se). The resource you are going to call dibs on is `scout-stage` and the server is `cg-vm1`.
1. `ssh <USER.NAME>@cg-vm1.scilifelab.se`
1. `sudo -iu hiseq.clinical`
1. `ssh localhost`
1. (optional) Find out which scout branch is currently deployed on cg-vm1: `podman ps`
1. Stop the service with current deployed branch: `systemctl --user stop scout.target`
1. Start the scout service with the branch to test: `systemctl --user start scout@<this_branch>`
1. Make sure the branch is deployed: `systemctl --user status scout.target`
1. After testing is done, repeat procedure at [https://pax.scilifelab.se/](https://pax.scilifelab.se), which will release the allocated resource (`scout-stage`) to be used for testing by other users.
</details>

<details>
<summary>Testing on hasta server (Clinical Genomics Stockholm)</summary>

**Prepare for testing**
1. `ssh <USER.NAME>@hasta.scilifelab.se`
1. Book your testing time using the Pax software. `us; paxa -u <user> -s hasta -r scout-stage`. You can also use the WSGI Pax app available at [https://pax.scilifelab.se/](https://pax.scilifelab.se).
1. (optional) Find out which scout branch is currently deployed on cg-vm1: `conda activate S_scout; pip freeze | grep scout-browser`
1. Deploy the branch to test: `bash /home/proj/production/servers/resources/hasta.scilifelab.se/update-tool-stage.sh -e S_scout -t scout -b <this_branch>`
1. Make sure the branch is deployed: `us; scout --version`
1. After testing is done, repeat the `paxa` procedure, which will release the allocated resource (`scout-stage`) to be used for testing by other users.
</details>

**How to test**:
1. how to test it, possibly with real cases/data

**Expected outcome**:
The functionality should be working
Take a screenshot and attach or copy/paste the output.

**Review:**
- [ ] code approved by
- [x] tests executed by CR

### CODE REVIEW

File: scout/server/blueprints/cases/controllers.py [hunk 1]
```diff
@@ -605,6 +607,10 @@ def case_report_content(store, institute_id, case_name):
             **DISMISS_VARIANT_OPTIONS,
             **CANCER_SPECIFIC_VARIANT_DISMISS_OPTIONS,
         }
+        data["genes_in_panels"] = store.panels_to_genes(
```
Comment (northwestwitch) [inline scout/server/blueprints/cases/controllers.py:610]: This step might take quite some time according to the size of the SVs, that's why I'm including it in the condition `if data["cancer"]:`. I'm open to suggestions here.

**Comment by northwestwitch** [comment]
Before:

image


After:

image

**Review by dnil — APPROVED** [review]
Clean and simple - let us test so the speed is acceptable.

We didn't specify it in the issue, but generally using the default panels is best: for the `master`-list cases, the number of case panel is pretty big.

**Comment by northwestwitch** [comment]
Ok, this is new:

One of the failed tests:

FAILED tests/server/blueprints/clinvar/test_clinvar_views.py::test_clinvar_update_submission - TypeError: quote() doesn't support 'encoding' for bytes
!!!!!!!!!!!!!!!!!!!!!!!!!! stopping after 1 failures !!!!!!!!!!!!!!!!!!!!!!!!!!!**

**Comment by northwestwitch** [comment]
A [new release of urlib](https://github.com/urllib3/urllib3/releases/tag/2.0.0a4) 17 hours ago is very suspicious

File: requirements.txt [hunk 2]
```diff
@@ -1,5 +1,5 @@
 # Everything needed in production
-werkzeug<2.2.3
+werkzeug<2.3
```
Comment (northwestwitch) [inline requirements.txt:2]: @dnil this should cover the situation with Docker

Comment (dnil) [reply requirements.txt:2]: Super; yes, and allow for stepping over 2.2.2 to make dependabot happy.

File: scout/server/blueprints/cases/controllers.py [hunk 3]
```diff
@@ -605,6 +607,14 @@ def case_report_content(store, institute_id, case_name):
...
+            panel_ids=[
+                panel["panel_id"]
+                for panel in case_obj.get("panels", [])
+                if panel.is_default is True
```
Comment (dnil) [inline scout/server/blueprints/cases/controllers.py:614]: 👍

**Comment by northwestwitch** [comment]
Tested again on stage and seems working. Merging!

**Pull request merged**